content/english/java-tictactoe/activity-5.md

```diff
@@ -3,5 +3,80 @@ title: "5. (Optional) AI Move"
 description: ""
 date: 2020-07-23T00:00:00Z
 weight: 6
-hidden: true
----
\ No newline at end of file
+hidden: false
+---
+{{% notice tip %}}
+#### This is a bonus activity!
+To continue, you should have completed the Tic-Tac-Toe program from the previous exercises! 
+{{% /notice %}}
+
+### Let's make the computer think
+Previously, we use a `Random` object to randomly generate a move for the computer, making the competitiveness of the computer is pretty low.
+
+In this exercise, we want to increase the difficulty of the game by making optimal decisions for the computer.
+
+We will do so by adding artificial intelligence into our program with the use the <b>Minimax</b> algorithm (a well-defined procedure that allows computers to solve problems).
+
+### Minimax Algorithm
+Minimax is an algorithm that is often used in two-player games to make optimal decisions for a player. 
+
+- The algorithm makes optimal decision for the maximizer by examining all the possible future game states based on the current game board. 
+- If there is a winner or tie, the game state will be given a score. (Score: computer wins (`positive value`), player wins (`negative value`), tie (`0`)).
+- The two players are labeled as <b>maximizer</b> and <b>minimizer</b> respectedly. In this case, the computer is the maximizer, and the player is the minimizer. 
+- The maximizer tries to get the highest score possible, and the minimizer tries to minimize the score of its opponent. 
+
+<img src="../images/minimax.png" height="500"/> 
+
+During the minimizing round, we pick the game state with the smallest score. And, in the maximizing round, we pick the game state with the largest score.
+
+If you follow along through the maximizing/minimizing rounds on the picture above, you will note that the best move for the computer is to place `"X"` at position 8.
```

Comment:
Do we need a picture with position labels?

Reply:
I think we established the grid is labeled with 1-9 in the last 4 exercise, but I will add a note :)